pkg/controller/installation/core_controller.go

```diff
@@ -1499,6 +1510,32 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	// Tell the status manager that we're ready to monitor the resources we've told it about and receive statuses.
 	r.status.ReadyToMonitor()
 
+	// BPF Upgrade without disruption:
+	// First, validate FelixConfig annotations before continuing.
+	err = bpfValidateAnnotations(felixConfiguration)
+	if err != nil {
+		r.status.SetDegraded(operator.ResourceValidationError, "Error validating resource", err, reqLogger)
```

Comment:
There's some nuance here as well - at this point in the code, we are not necessarily going to update the FelixConfiguration - e.g., if the value is already what it needs to be.

Probably want to move this check within `setBPFEnabled`, since really this function is meant to tell us whether updating that field is allowed or not, and including the check there ensures we can't accidentally set the field without checking the annotation.

Reply:
Done - thanks